sympy/series/series.py

```diff
@@ -10,3 +13,151 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
+    -(x**2*exp(x) - E)**3*(-14*exp(3) - 14*E + 7 + 21*exp(2) + 7*exp(4))*exp(-4)/54 - 7*(x**2*exp(x) - E)**2*exp(-2)/54
+    + (x**2*exp(x) - E)*exp(-1)/3 + 1
+
+    Here the number of terms required is, by default, 3.
+
+    >>> lagrange_inversion_theorem(eq, x, 1)
+    -7*(x**2*exp(x) - E)**2*exp(-2)/54 + (x**2*exp(x) - E)*exp(-1)/3 + 1
+
+    Here the point with respect to which the expansion is found is changed from x=1 to x=2.
+
+    >>> lagrange_inversion_theorem(eq, x, 2)
+    -7*(x**2*exp(x) - 4*exp(2))**2*exp(-4)/512 + (x**2*exp(x) - 4*exp(2))*exp(-2)/8 + 2
+
+    >>> eq = exp(x)
+    >>> lagrange_inversion_theorem(eq, x, 2)
+    -(exp(x) - exp(2))**2*exp(-4)/2 + (exp(x) - exp(2))*exp(-2) + 2
+
+    Here the number of terms and the point with respect to which the inverse expansion will be calculated are both
+    not provided explicitly.
+
+    >>> lagrange_inversion_theorem(eq, x)
+    -(exp(x) - 1)**2/2 + exp(x) - 1
+
+    Note
+    ====
+
+    There is a straightforward derivation using complex analysis and contour integration;
+    the complex formal power series version is a consequence of knowing the formula for polynomials,
+    so the theory of analytic functions may be applied. Actually, the machinery from analytic
+    function theory enters only in a formal way in this proof, in that what is really needed is
+    some property of the formal residue, and a more direct formal proof is available.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Lagrange_inversion_theorem
+    .. [2] http://mathworld.wolfram.com/LagrangeInversionTheorem.html
+    .. [3] http://www.cfm.brown.edu/people/dobrush/am33/Mathematica/lit.html
+    """
+
+    try:
+        not eq.free_symbols
+    except AttributeError:
+        raise ValueError("The function provided is not analytic i.e. f' = 0. Lagrange's inversion theorem requires "
+                         "that the function be analytic")
+
+    if len(eq.free_symbols) == 1:
+        if x not in eq.free_symbols:
+            raise ValueError("The function variable provided does not match the equation variable")
+
+    w = x
+    eq = eq.subs(x, w)
+    z = eq
+
+    z_diff = diff(eq, w)
+
+    if z_diff.subs(w, a) == 0 or not (z_diff.subs(w, a)).is_real:
+        raise ValueError("The first derivative of the function is zero at the point 'a' provided i.e. f'(a) = 0. "
+                         "Lagrange's inversion theorem requires that the derivative of the function be non zero at "
+                         "the point 'a'.")
+
+    f_a = z.subs(w, a)
+    result = a
+
+    if n == 1:
+        return result
+
+    a_1 = ((w - a)/(z - f_a))
+    l = limit(a_1, w, a)
+    result = result + l*(z - f_a)
+
+    if n == 2:
+        return result
+
+    a_2 = diff(((w - a)/(z - f_a))**2, w, 1)
+
+    for i in range(2, n):
+        l = limit(a_2, w, a)
+        result = result + (((z - f_a) ** i) / factorial(i)) * l
+        a_2 = (i+1) * ((w - a / z - f_a) ** i) * a_2
```

Comment:
Also, the way how `n` binds to the polynomial degree may had changed as in the above example.
Now, it evaluates until it reaches 4 instead of 5.
However, I also think that recent change can be more compatible with the other implementation, `series`.

But `series` also returns a truncated order term appended with the series expansion, like `O((x - 1)**4, (x, 1))`

<img src="https://user-images.githubusercontent.com/34944973/52969175-cadb9080-33f2-11e9-8ef8-b673d7950a1b.png" width="500px">
Can you add the feature to this function?

Reply:
I'm not sure if I understand what you're trying to say here.
The reason why n=1 always return an integer is because of the formula of Lagrange inversion. It contains the term `a`. This is what is returned as the first term in the expansion and is counted as one of the terms. 
If I've misunderstood what you were trying to say here, please let me know.